src/_pytest/python.py

```diff
@@ -1000,49 +1023,59 @@ def _resolve_arg_ids(
         :rtype: List[str]
         :return: the list of ids for each argname given
         """
-        idfn = None
-        if callable(ids):
+        if ids is None:
+            idfn = None
+            ids_ = None
+        elif callable(ids):
             idfn = ids
-            ids = None
-        if ids:
-            func_name = self.function.__name__
-            ids = self._validate_ids(ids, parameters, func_name)
-        ids = idmaker(argnames, parameters, idfn, ids, self.config, item=item)
-        return ids
-
-    def _validate_ids(self, ids, parameters, func_name):
+            ids_ = None
+        elif not ids:
+            # Special case: https://github.com/pytest-dev/pytest/issues/1849
+            idfn = None
+            ids_ = []
+        else:
+            idfn = None
+            ids_ = self._validate_ids(ids, parameters, self.function.__name__)
+        return idmaker(argnames, parameters, idfn, ids_, self.config, item=item)
+
+    def _validate_ids(
+        self,
+        ids: Iterable[Union[None, str, float, int, bool]],
+        parameters: typing.Sequence[ParameterSet],
+        func_name: str,
+    ) -> List[Union[None, str]]:
         try:
-            len(ids)
```

Comment:
I had to refactor a bit to make it amenable to mypy. It's meant to be equivalent.